src/loader.js

```diff
@@ -146,15 +147,99 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
+    var iterator;
+    if (__global.Map && __global.Map.prototype && __global.Map.prototype.entries && __global.Map.prototype.keys
+         && __global.Map.prototype.values && __global.Map.prototype.get && __global.Map.prototype.set
+         && __global.Map.prototype.has && __global.Map.prototype.delete)
+      registry._registry.registryMap = new __global.Map();
+    else
+      registry._registry.registryMap = new InternalMapPolyfill();
+    // 4.4.2
+    if (__global.Symbol && __global.Symbol.iterator)
+      registry[__global.Symbol.iterator] = mapPolyfillEntriesIterator.bind(registry._registry.registryMap);
+    return registry;
+  }
+
+  // 4.2
+  function Registry() {
+    throw new Error('A registry may only be created when creating a loader');
+  }
```

Comment:
I was under the impression that the Registry constructor actually would be public because of the language in https://whatwg.github.io/loader/#registry-constructor. The reason for that is because it talks about the Registry constructor as a separate thing from the %Registry% intrinsic object. Admittedly, though, I don't have any real training / experience on how to read these specs and so maybe I am misinterpreting. Could you shed some light on how the verbage in that link should be interpreted?